### PR TITLE
Add using for some types to work around compiler crash SR70253

### PR DESCRIPTION
**Pull request by hjyamauchi**
### Description

Reference some types so that they are included in the CXX module.

### Testing

Built Arc with cmake in debug mode. 

### Type of Change
Place an `x` the applicable box:
- [x] Bug fix. Add the issue # below if applicable. WIN-971
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

**Comment by brianmichel** [comment]
I'm going to defer to @compnerd since my C++ is weak

File: firestore/src/include/firebase/firestore/map_field_value.h [hunk 1]
```diff
@@ -31,6 +31,13 @@ using MapFieldValue = std::unordered_map<std::string, FieldValue>;
 /** @brief A map of `FieldValue`s indexed by field paths. */
 using MapFieldPathValue = std::unordered_map<FieldPath, FieldValue>;
 
+// Reference the following types so that they are included in the CXX
+// module. A workaround for deserialization cross reference errors
+// WIN-971.
```
Comment (tristanlabelle) [inline firestore/src/include/firebase/firestore/map_field_value.h]: For upstreamability, we should avoid referencing an internal ticket, but we could reference a bug on an open-source Swift repo.

**Review by tristanlabelle — APPROVED** [review]

File: firestore/src/include/firebase/firestore/map_field_value.h (hunk 1 above)
Comment (compnerd) [reply firestore/src/include/firebase/firestore/map_field_value.h]: Just re-enforcing what @tristanlabelle already stated.  Also, please wrap this in a `#if defined(__swift__)` as we would like these patches to be upstreamable.

Comment (hjyamauchi) [reply firestore/src/include/firebase/firestore/map_field_value.h]: @tristanlabelle do you mean a bug on apple/swift?

Comment (compnerd) [reply firestore/src/include/firebase/firestore/map_field_value.h]: @hjyamauchi yeah; when this was in JIRA, we would reference them as SR\d+.

Comment (hjyamauchi) [reply firestore/src/include/firebase/firestore/map_field_value.h]: Done

**Comment by hjyamauchi** [comment]
PTAL

**Review by compnerd — APPROVED** [review]

**Review by jeffdav — APPROVED** [review]
I've verified that this, in conjunction with the other workaround, does avoid the crash!

**Pull request merged**